R/set_tech_trajectories.R

```diff
@@ -430,3 +442,214 @@ filter_negative_late_and_sudden <- function(data_with_late_and_sudden, log_path)
 
   return(data_with_late_and_sudden)
 }
+
+#' Defines which scenario values to use for the production trajectory after a
+#' litigation event in the LRISK stress test.
+#'
+#' @description
+#' Picks the corresponding values from the original scenario
+#' column indicated in the input and has the option to include PACTA based
+#' production forecast for the first few years of the late & sudden
+#' trajectory. Similarly, it is possible to define another input scenario
+#' in case the company is already aligned after the production forecast.
+#' If the production forecast is included, the trajectory after the end of
+#' the production forecast is offset by the initial production forecast
+#' so that the remainder of the late & sudden trajectory now is a parallel
+#' shift of the original scenario values. If not included, the trajectories
+#' replicate externally provided scenario trajectories at least until the
+#' year of the policy shock.
+#' Trajectories are calculated for each company by sector, scenario_geography,
+#' technology, year.
+#' If no "id" or "company_name" are provided, the calculation switches to
+#' portfolio/technology level.
+#' @inheritParams report_company_drops
+#' @param data A dataframe that contains the scenario data prepared until the
+#'   step after the baseline trajectories are calculated.
+#' @param litigation_scenario Character. A string that indicates which
+#'   of the scenarios included in the analysis should be used to set the
+#'   technology trajectories post litigation event.
+#' @param shock_scenario A dataframe that contains information about the
+#'   transition scenario, specifically the shock year and, duration of the
+#'   shock and the name of the shock scenario
+#' @param litigation_scenario_aligned Character. A string that indicates which
+#'   of the scenarios included in the analysis should be used to set the
+#'   technology trajectories post litigation event in case the company is
+#'   aligned after the forecast period.
+#' @param start_year Numeric. A numeric vector of length 1 that contains the
+#'   start year of the analysis.
+#' @param end_year Numeric. A numeric vector of length 1 that contains the
+#'   end year of the analysis.
+#' @param analysis_time_frame Numeric. A vector of length 1 indicating the number
+#'   of years for which forward looking production data is considered.
+#'
+#' @return data frame
+set_litigation_trajectory <- function(data,
+                                      litigation_scenario,
+                                      shock_scenario,
+                                      litigation_scenario_aligned,
+                                      start_year = 2020,
```

Comment:
where do these hardcoded defaults come from and are they a good idea?

Reply:
no, bad idea. removed